### DISCUSSION TITLE
Shared history across multiple editor instances but not related to collaboration.

### DISCUSSION THREAD

**Discussion by githubbob42** [ueberdosis/tiptap#3650] [General Questions]
Hi all,

We are writing a PDF editor with TipTap v2 and React and it works pretty darned good (thanks TipTap - y'all are awesome).  

However, because of the nature of how PDF docs are constructed, the header, body and footer are considered separate sections (for lack of a better term) with each section essentially being absolutely positioned with its own margins, etc. (PDFs don't "flow" like markup languages).  So in our editor, we have 3 separate editor instances - for the header, body and footer.

![image](https://user-images.githubusercontent.com/2529300/214143882-60a1b2d0-601e-484b-89e9-2846ed338ebf.png)

Unfortunately, each editor maintains it's own history, so clicking `ctrl-z` performs an undo only in the currently selected editor instance and gives unexpected/unpredictable results to the end user.

https://user-images.githubusercontent.com/2529300/214176330-c7b160cc-d067-450d-a42e-a9827a5028c7.mp4

Is there a way to share history across multiple editor instances?  

The only thing I can think of is to hook into the `onUpdate` event on each editor and create a single history stack to track updates across the editors and override `ctrl-z` so that it calls `editor.commands.undo()` for the editor on the current history element on the stack and pops them off in turn.  

Sounds fairly easy on the surface but there is a lot going on under the hood in `prosemirror-history`; like keeping track of undo and redo histories (branches), `depth`, `newGroupDelay`, etc on top of the fact that we can drag elements (words, paragraphs, tables, etc) between the editors.

Any ideas would be appreciated.  Thanks in advanced...

**Comment by marczi** [comment]
Dear @githubbob42 

Did you manage to solve this? I have the same problem, but i think i should implement it at ProseMirror level, maybe it should be a custom history plugin

Thanks

**Comment by markbenliyan** [comment]
+1, has this been solved?

**Comment by Gianthard-cyh** [comment]
Curious if anyone has figured out a solution for this yet.
